src/cli/compiler.js

```diff
@@ -51,21 +61,143 @@ class Compiler {
         : path.join(this.options.outputDir, 'subgraph.yaml')
 
     this.logger.info('')
-    this.logger.info(chalk.green('Completed'))
+    this.logger.info(chalk.green('Build completed'))
     this.logger.info('')
     this.logger.info('%s %s', chalk.bold(chalk.blue('Subgraph:')), hashOrFilename)
     this.logger.info('')
   }
 
   loadSubgraph() {
     try {
-      this.logger.step('Load subgraph:', this.options.subgraphManifest)
       return Subgraph.load(this.options.subgraphManifest)
     } catch (e) {
       this.logger.fatal('Failed to load subgraph:', e)
     }
   }
 
+  getFilesToWatch() {
+    try {
+      let allFiles = []
+      let subgraph = this.loadSubgraph()
+
+      // Add all file paths specified in manifest
+      allFiles.push(subgraph.getIn(['schema', 'file']))
+      subgraph.get('dataSources').map(dataSource => {
+          allFiles.push(dataSource.getIn(['mapping', 'file']))
+          dataSource.getIn(['mapping', 'abis']).map(abi => {
+            allFiles.push(abi.get('file'))
+          })
+      })
+
+      // Make paths absolute
+      let allAbsolutePaths = allFiles.map(file => {
+        return path.resolve(this.sourceDir, file)
+      })
+      return allAbsolutePaths
+    } catch(e) {
+      this.logger.fatal('Failed to parse subgraph file locations:', e)
+    }
+  }
+
+  watchAndCompile() {
+    let compiler = this
+    if (cluster.isMaster) {
+      let worker = cluster.fork()
+      worker.send({msgFromMaster: 1});
+
+      cluster.on('exit', function() {
+        cluster.fork()
+      })
+    }
+
+    if (cluster.isWorker) {
+      // Use master message to kick off compile
+      // preventing infinite error loop
+      process.on('message', function(_) {
+        compiler.compile()
+      });
+      let watchedFiles = this.getFilesToWatch()
+
+      // Initialize watcher that reruns compiler
+      let watcher = chokidar.watch(this.options.subgraphManifest, {
+        persistent: true,
+        ignoreInitial: true,
+        atomic: 500
+      })
+      watcher
+        .on('ready', function () {
+          compiler.logger.info('')
+          compiler.logger.info(chalk.grey('Watching relevant manifest files'))
+        })
+        .on('change', changedFile => {
+          compiler.logger.info('')
+          compiler.logger.info('%s %s',
+            chalk.grey('File change detected: '),
+            compiler.displayPath(changedFile)
+          )
```

Comment:
I personally like having the file watcher updates as `logger.info, but it may make sense to hide in default setting .  Strong opinions?